csrc/ops/alias.h

```diff
@@ -30,6 +30,12 @@ TORCH_CUDA_CU_API TensorView* reshape(
     const std::vector<int64_t>& original_sizes,
     const std::vector<int64_t>& new_sizes);
 
+//! Dynamic version of reshape. The number of dimensions cannot
+//! change, but the actual sizes of the dimensions can be symbolic.
+TORCH_CUDA_CU_API TensorView* reshape(
```

Comment:
I didn't realize that we have a requirement on tensor staying the same rank here... that's an interesting restriction.

Reply:
I figure we can fake it on the frontend... so that leave me wondering why we are not doing this on the backend side?!?

i.e. We can have a `reshape` to get a size-1 dimension and then squeeze. (or unsqueeze if we are expanding).

Reply:
NVM. I think it's just the comment here.

> Dynamic version of reshape. The number of dimensions cannot change, but the actual sizes of the dimensions can be symbolic.

nitpick: this sounds like that we are requiring output to be the same rank as with the input, which I think is not the case at all.

Reply:
Updated the comment.